app/resources/inventory.py

```diff
@@ -6,6 +6,16 @@
 from vars import game_info
 
 
+@m.user_endpoint(path=["inventory", "trade"], requires=trade_requirements)
+def trade(data: dict, user: str) -> dict:
```

Comment:
You use `data["target"]` pretty often. Just out source this into a variable:
```suggestion
def trade(data: dict, user: str) -> dict:
    target: Optional[str] = data["target"]
```